src/components/ha-date-range-picker.ts

```diff
@@ -36,12 +37,20 @@ export class HaDateRangePicker extends LitElement {
 
   @property({ type: Boolean }) private _hour24format = false;
 
+  @property({ type: String }) private _rtlDirection = "ltr";
+
   protected updated(changedProps: PropertyValues) {
     if (changedProps.has("hass")) {
       const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
       if (!oldHass || oldHass.language !== this.hass.language) {
         this._hour24format = this._compute24hourFormat();
       }
+      if (
+        !oldHass ||
+        computeRTLDirection(oldHass) !== computeRTLDirection(this.hass)
```

Comment:
Just compare the languages, you are now doing 2 `computeRTLDirection` extra.

Reply:
But lang can change and RTLDirection may not change. It is inefficient however I though re-rendering was the thing to avoid at almost all cost??

Reply:
I Joint both checks